app/src/main/java/com/example/filemanager/AllFilesFragment.kt

```diff
@@ -28,16 +28,33 @@ class AllFilesFragment : Fragment(), ItemClickListener {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        setHasOptionsMenu(true)
         viewModel = ViewModelProvider(this, AllFilesViewModelFactory(applicationContext()))
             .get(AllFilesViewModel::class.java)
         adapter = ListAdapter(this)
+
+        requireActivity().onBackPressedDispatcher.addCallback(this, object
+            : OnBackPressedCallback(true) {
```

Comment:
can inline the `OnBackPressedCallback` with `object`

`requireActivity().onBackPressedDispatcher.addCallback(this, object : OnBackPressedCallback(true) {`